dallinger/command_line/__init__.py

```diff
@@ -523,49 +527,61 @@ def _current_hits(service, app):
     flag_value=True,
     help="Look for HITs in the MTurk sandbox rather than the live/production environment",
 )
-def hits(app, sandbox):
-    """List all HITs for the user's configured MTurk request account,
-    or for a specific experiment id.
-    """
+@click.option("--recruiter", default="mturk", help="Experiment id")
+def hits(app, sandbox, recruiter):
+    """List all HITs for the recruiter account or for a specific experiment id."""
     if app is not None:
         verify_id(None, "--app", app)
-    formatted_hit_list = []
-    dateformat = "%Y/%-m/%-d %I:%M:%S %p"
-    for h in _current_hits(_mturk_service_from_config(sandbox), app):
-        title = h["title"][:40] + "..." if len(h["title"]) > 40 else h["title"]
-        description = (
-            h["description"][:60] + "..."
-            if len(h["description"]) > 60
-            else h["description"]
-        )
-        formatted_hit_list.append(
-            [
-                h["id"],
-                title,
-                h["annotation"],
-                h["status"],
-                h["created"].strftime(dateformat),
-                h["expiration"].strftime(dateformat),
-                description,
-            ]
+
+    get_recruiter(recruiter).hits(app, sandbox)
+
+
+@dallinger.command()
+@click.option("--hit_id", default=None, help="MTurk HIT ID")
+@click.option(
+    "--sandbox",
+    is_flag=True,
+    flag_value=True,
+    help="Look for HITs in the MTurk sandbox rather than the live/production environment",
+)
+@click.option("--recruiter", default="mturk", help="Experiment id")
+def hit_details(hit_id, sandbox, recruiter):
+    """Print the details for a specific HIT is for a recruiter."""
+    details = get_recruiter(recruiter).hit_details(hit_id, sandbox)
+    print(json.dumps(details, indent=4, default=str))
+
+
+@dallinger.command()
+@click.option("--hit_id", default=None, help="MTurk HIT ID")
+@click.option(
+    "--sandbox",
+    is_flag=True,
+    flag_value=True,
+    help="Look for HITs in the MTurk sandbox rather than the live/production environment",
+)
+@click.option("--recruiter", default="mturk", help="Experiment id")
+@click.option(
+    "--qualification_path",
+    default=None,
+    help="Filename/path for the qualification file",
+)
+def copy_qualifications(hit_id, sandbox, recruiter, qualification_path):
+    """Copy qualifications from an existing HIT ID."""
+    recruiter = get_recruiter(recruiter)
+    if qualification_path is None:
+        qualification_path = recruiter.default_qualification_name
+    assert qualification_path.endswith(
+        ".json"
+    ), "Qualification path must be a json file"
+    if exists(qualification_path):
+        overwrite = query_yes_no(
```

Comment:
I would probably just use `click.confirm` instead of adding `query_yes_no` to the codebase.

Reply:
Actually, I avoided this because this forces the user to always confirm. I only want confirmation in case the file already exists.

Reply:
You can just write

``` py
if exists(qualification_path) and click.confirm(...)
```

though?

Reply:
Yep, I implemented a variant of that.